google/cloud/aiplatform/utils.py

```diff
@@ -318,20 +337,79 @@ def __init__(
             Optional. Client credentials to pass to client.
         """
 
-        self._client_class = client_class
         self._credentials = credentials
         self._client_options = client_options
         self._client_info = client_info
-
-    def __getattr__(self, name: str) -> Any:
-        """Instantiates client and returns attribute of the client."""
-        temporary_client = self._client_class(
+        self._clients = {
+            version: client
+            for version, client in self._version_map
+        }
+    def _temporary_client(self, version) -> AiPlatformServiceClient:
+        temporary_client = self._clients[version](
             credentials=self._credentials,
             client_options=self._client_options,
             client_info=self._client_info,
         )
-        return getattr(temporary_client, name)
+        return temporary_client
+
+    def __getattr__(self, name: str) -> Any:
+        """Instantiates client and returns attribute of the client."""
+        return getattr(self._temporary_client(self._default_version), name)
+
+    def select_version(self, version: str) -> AiPlatformServiceClient:
+        return self._temporary_client(version)
```

Comment:
One potential issue with this approach is that we are not enforcing that the client is temporary like we are through the `__getattr__` path.